### DISCUSSION TITLE
Multiple controls, single state factor

### DISCUSSION THREAD

**Discussion by patasab** [infer-actively/pymdp#112] [Q&A] [answered]
Hi,

Thank you very much for this package; it is a valuable contribution to the field.

I have a question about the possibility of having two actions and a single factor whose transition depends on both actions.
I tried adding a fourth dimension to the B matrix to include the second control, but it caused errors. Another way that I'm pretty sure could work is to define a single control with all the configurations of actions. In my case, it is entirely feasible since I have only two possible actions for each control. However, I was wondering if there is a more general and proper way to fulfill this requirement.

Thank you in advance.

Kind regards,
Sabrina

**Comment by conorheins** [comment, marked as answer]
Hi @patasab,
This is an interesting idea. 
Over the past few days (on [this branch](https://github.com/infer-actively/pymdp/tree/sparse_likelihoods_111)) I've been working on something very similar to what you're asking about, but not quite the same.

Once I merge that branch (which you can already locally clone / dev install if you're interested in using it), then you will be able to specify hidden state factor transitions that depend on _other factors_. However, each hidden state factor will still have to depend on a single control factor, which is distinct from what you're imagining.

With that being said, because a single control factor determines the dynamics of its own hidden state factor, and that hidden state factor may itself influence another hidden state factor in this new version of pymdp with 'interactable' hidden state factors, then in some ways control factors may "jointly determine" the dynamics of another hidden state factor, but only _via_ their influence on their own proper hidden state factor. Do you think you could use this slightly-indirect approach to create the sort of dynamics you're imagining?

If not, then I would encourage you to simply go with your second suggestion: "define a single control with all the configurations of actions".
This is probably the simplest way to move forward, since 1) you can do this with the existing master branch of `pymdp` and 2) your control space is quite small so the combinatorics of your action space is quite manageable (2 actions x 2 control factors = 4 possible actions, i.e. all possible combinations of the two actions).

**Reply by patasab** [reply]
Thank you, @conorheins, for your reply. I'm glad to hear that you're working on that extension, as it can be very beneficial in many situations.
However, for my current project, I find the second solution to be more convenient. I can't really define a factor for each control, but "unfolding" the controls seems like a more natural solution for the problem at hand. Thanks :-)
